File: baekjoon/알파벳_다이아몬드.js
const [n, r1, c1, r2, c2] = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split(" ")
  .map(Number);

const mid = n - 1;
const end = mid * 2;
const baseChar = "a".charCodeAt(0);

let res = "";

for (let i = r1; i <= r2; i++) {
  for (let j = c1; j <= c2; j++) {
    const dist =
      Math.abs((i % (end + 1)) - mid) + Math.abs((j % (end + 1)) - mid);
    if (dist >= n) {
      res += ".";
      continue;
    }
    res += String.fromCharCode(baseChar + (dist % 26));
  }
  res += "\n";
}

console.log(res);

File: baekjoon/알파벳_다이아몬드.py
n, r1,c1, r2,c2 = list(map(int, input().split()))

for i in range(r1, r2+1):
    res = ""
    for j in range(c1, c2 +1):
        dist = abs(i % (n * 2 -1) - n + 1) + abs(j % (n * 2 -1) - n + 1)
        if (dist >= n):
            res += "."
            continue
        res += str(chr(dist + 97))
    print(res)